substrate/frame/balances/src/lib.rs

```diff
@@ -730,22 +683,6 @@ pub mod pallet {
 			}
 		}
 
-		/// Alias for `transfer_allow_death`, provided only for name-wise compatibility.
-		///
-		/// WARNING: DEPRECATED! Will be released in approximately 3 months.
-		#[pallet::call_index(7)]
-		#[pallet::weight(T::WeightInfo::transfer_allow_death())]
-		pub fn transfer(
-			origin: OriginFor<T>,
-			dest: AccountIdLookupOf<T>,
-			#[pallet::compact] value: T::Balance,
-		) -> DispatchResult {
-			let source = ensure_signed(origin)?;
```

Comment:
The only thing that we could do is to always return an error with a message that the other call should be used.  
But its unclean i think…

Reply:
I feel like we missed an opportunity to mark this function as `#[deprecated]`. Maybe this would've clued people in that this extrinsic is going away...

Reply:
There was a deprecation message, tagging `#[deprecated]` wouldn't have helped much here anyway as external calls don't get notified of it. See https://github.com/paritytech/polkadot-sdk/issues/182#issuecomment-1691684159